### PR TITLE
join: Add automatic sort

### PR DESCRIPTION
**Pull request by mattnibs**
Add functionality to join operator to automatically sort any unsorted input.

### CODE REVIEW

File: runtime/op/join/join.go [hunk 1]
```diff
@@ -33,11 +34,25 @@ type Op struct {
...
+		if err != nil {
+			return nil, err
+		}
+	}
+	if !rightSorted || leftOrder != rightOrder {
+		right, err = sort.New(octx, right, []expr.Evaluator{rightKey}, rightOrder, true)
+		if err != nil {
+			return nil, err
+		}
+	}
```
Comment (nwt) [inline runtime/op/join/join.go:55]: Something's wrong here.

Suppose you create two files.
```console
$ cat <<EOF >a.zson
{a:1}
{a:2}
EOF
$ cat <<EOF >b.zson
{b:2}
{b:1}
EOF
```

Joining those files on `a=b` should always produce two output values.  But in these cases it doesn't.
```console
$ zq 'file a.zson | join (file b.zson | sort -r b) on a=b'
{a:2}
$ zq 'file a.zson | sort a | join (file b.zson | sort -r b) on a=b'
{a:2}
$ zq 'file a.zson | sort -r a | join (file b.zson) on a=b'
{a:1}
$ zq 'file a.zson | sort -r a | join (file b.zson | sort b) on a=b'
{a:1}
```

Comment (mattnibs) [reply runtime/op/join/join.go:55]: This is v strange, thanks for digging into this.

Comment (mattnibs) [reply runtime/op/join/join.go:55]: Actually a much better test as well will use this.

**Review by nwt — COMMENTED** [review]
You should update the Description section of docs/language/operators/join.md so that it doesn't say inputs must be sorted.

**Review by nwt — COMMENTED** [review]
And docs/tutorial/join.md also contains references to sorting that should be updated.

**Review by nwt — APPROVED** [review]
:shipit: when CI is green.

**Review by philrz — APPROVED** [review]
Docs changes look good too! 👍

**Pull request merged**